bot/cogs/duck_pond.py

```diff
@@ -0,0 +1,174 @@
+                    if reaction.emoji == "🦆":
+                        duck_count += 1
+                        duck_reactors.append(user.id)
+        return duck_count
+
+    async def relay_message_to_duck_pond(self, message: Message) -> None:
+        """Relays the message's content and attachments to the duck pond channel."""
+        clean_content = message.clean_content
+
+        if clean_content:
+            await self.send_webhook(
+                content=message.clean_content,
+                username=message.author.display_name,
+                avatar_url=message.author.avatar_url
+            )
+
+        if message.attachments:
+            try:
+                await send_attachments(message, self.webhook)
+            except (errors.Forbidden, errors.NotFound):
+                e = Embed(
+                    description=":x: **This message contained an attachment, but it could not be retrieved**",
+                    color=Color.red()
+                )
+                await self.send_webhook(
+                    embed=e,
+                    username=message.author.display_name,
+                    avatar_url=message.author.avatar_url
+                )
+            except discord.HTTPException:
+                log.exception(f"Failed to send an attachment to the webhook")
+
+        await message.add_reaction("✅")
+
+    @Cog.listener()
+    async def on_raw_reaction_add(self, payload: RawReactionActionEvent) -> None:
+        """
+        Determine if a message should be sent to the duck pond.
+
+        This will count the number of duck reactions on the message, and if this amount meets the
+        amount of ducks specified in the config under duck_pond/threshold, it will
+        send the message off to the duck pond.
+        """
+        # Is the emoji in the reaction a duck?
+        if payload.emoji.is_custom_emoji():
+            if payload.emoji.id not in constants.DuckPond.custom_emojis:
+                return
+        elif payload.emoji.name != "🦆":
+            return
+
+        channel = discord.utils.get(self.bot.get_all_channels(), id=payload.channel_id)
+        message = await channel.fetch_message(payload.message_id)
+        member = discord.utils.get(message.guild.members, id=payload.user_id)
+
+        # Is the member a human and a staff member?
+        if not self.is_staff(member) or member.bot:
+            return
+
+        # Does the message already have a green checkmark?
+        if await self.has_green_checkmark(message):
+            return
+
+        # Time to count our ducks!
+        duck_count = await self.count_ducks(message)
+
+        # If we've got more than the required amount of ducks, send the message to the duck_pond.
+        if duck_count >= constants.DuckPond.threshold:
+            await self.relay_message_to_duck_pond(message)
+
+    @Cog.listener()
+    async def on_raw_reaction_remove(self, payload: RawReactionActionEvent) -> None:
```

Comment:
Is this really needed? It only cares about the bot's checkmark and ignores those of other users. Only the bot can remove its own reaction (which it would never do since we didn't program it to do that), right?

Reply:
You're right. I tried but didn't see how do it at first.

Reply:
might be overkill. I don't mind if we keep it or not.

Reply:
Already done so may as well keep it.